test/resolve/metadataResolver.test.ts

```diff
@@ -979,5 +979,47 @@ describe('MetadataResolver', () => {
         expect(result).to.deep.equal([]);
       });
     });
+
+    it('should filter out empty directories when resolving components', () => {
+      const resolver = testUtil.createMetadataResolver([
+        {
+          dirPath: 'lwc',
+          children: ['myComponent', 'emptyComponent'],
+        },
+        {
+          dirPath: 'lwc/myComponent',
+          children: ['myComponent.js', 'myComponent.js-meta.xml'],
+        },
+        {
+          dirPath: 'lwc/emptyComponent',
+          children: [], // Empty directory
+        },
+      ]);
+
+      const expectedComponent = new SourceComponent({
+        name: 'myComponent',
+        type: registry.types.lightningcomponentbundle,
+        content: 'lwc/myComponent',
```

Comment:
probably need the `path.join` on these paths to fix windows UTs, otherwise looks good once tests pass